src/track/serato/beatgrid.cpp

```diff
@@ -511,10 +513,12 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
     // Finally, create the terminal marker.
     const double positionSecs =
             streamInfo.getSignalInfo().frames2secs(
-                    streamInfo.getSignalInfo().samples2frames(
-                            currentBeatPositionSamples)) -
+                    static_cast<SINT>(currentBeatPositionFrames)) -
             timingOffsetSecs;
-    const double bpm = pBeats->getBpmAroundPosition(currentBeatPositionSamples, 1);
+    const double bpm = pBeats->getBpmAroundPosition(
+            streamInfo.getSignalInfo().frames2samples(
+                    static_cast<SINT>(currentBeatPositionFrames)),
```

Comment:
and here also no SINT cast

Reply:
This one will still trigger the assertion.

Reply:
No. *Samples* needs to be a multiple of 2, but frames don't.

Here's the complete implementation of this method:
```
    // Conversion: #frames / frame offset -> #samples / sample offset
    SINT frames2samples(SINT frames) const {
        DEBUG_ASSERT(getChannelCount().isValid());
        return frames * getChannelCount();
    }
```

Reply:
Ah right, we have no assertion this way around.